core/app/services/spree/cart/change_currency.rb

```diff
@@ -0,0 +1,27 @@
+module Spree
+  module Cart
+    class ChangeCurrency
+      prepend Spree::ServiceModule::Base
+
+      def call(order:, new_currency:)
+        return failure('Currency not supported') unless supported_currency?(order, new_currency)
+
+        result = order.update!(currency: new_currency) rescue false
```

Comment:
I noticed that both `update` and `update!` throw exceptions sometimes. For example when one of the products in cart is not available in new currency. This also rolls-back the update, so I implemented it in a way that we treat exception as a failure. Let me know if that makes sense